query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -382,9 +383,13 @@ private boolean isEquals(Set<String> values, Expression expression) {
         request = originalRequest;
       }
 
-      Entry<String, Params> pql =
+      Pair<Map<String ,Integer>,Entry<String, Params>> pair =
           request2PinotSqlConverter.toSQL(
               executionContext, request, executionContext.getAllSelections());
+      Map<String,Integer> selectionMap = pair.getLeft();
```

Comment:
I am wondering what is the need to return this selectionMap from the `toSQL` method. 
If you see, the selectionMap is built using `executionContext.getAllSelections()` param, which is already available here.

Reply:
Can be done. But [this](https://github.com/hypertrace/query-service/blob/895ba1b4e99c456873bfd36fd2662af25b1084e2/query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java#L262) may be duplicated.

Reply:
Which should be fine I guess